virtool/hmm/api.py

```diff
@@ -27,167 +34,206 @@
 routes = Routes()
 
 
-@routes.get("/hmms")
-async def find(req):
-    """
-    Find HMM annotation documents.
+@routes.view("/hmms")
+class HMMView(PydanticView):
 
-    """
-    db = req.app["db"]
+    async def get(self) -> r200[List[HMMMinimal]]:
+        """
+        Find HMM annotation documents.
 
-    term = req.query.get("find")
+        Status Codes:
+            200: Successful operation
+        """
+        db = self.request.app["db"]
 
-    db_query = dict()
+        term = self.request.query.get("find")
 
-    if term:
-        db_query.update(compose_regex_query(term, ["names"]))
+        db_query = {}
 
-    data = await paginate(
-        db.hmm,
-        db_query,
-        req.query,
-        sort="cluster",
-        projection=PROJECTION,
-        base_query={"hidden": False},
-    )
+        if term:
+            db_query.update(compose_regex_query(term, ["names"]))
 
-    data["status"] = await virtool.hmm.db.get_status(db)
+        data = await paginate(
+            db.hmm,
+            db_query,
+            self.request.query,
+            sort="cluster",
+            projection=PROJECTION,
+            base_query={"hidden": False},
+        )
 
-    return json_response(data)
+        data["status"] = await virtool.hmm.db.get_status(db)
 
+        return json_response(data)
 
-@routes.get("/hmms/status")
-async def get_status(req):
-    """
-    Get the status of the HMM data. Contains the following fields:
+    @permissions(Permission.modify_hmm)
+    async def delete(self) -> Union[r204, r403]:
+        """
+        Delete all unreferenced HMMs and hide the rest.
```

Comment:
Let's update this since it is going to face users of the API in the documentation.

Right now it is describing an implementation detail instead of the behaviour as it affects the user.
```
Delete all installed HMM data.

This won't break analyses that reference the installed HMM data.
```